web-service-consumer/1.6/modules/ROOT/pages/edits/web-service-consumer-studio.adoc

```diff
@@ -0,0 +1,282 @@
+
+The following XML example shows a DataWeave expression inside the `headers` parameter:
+[source,xml,linenums]
+----
+<wsc:consume config-ref="config" operation="addClients">
+    <wsc:message>
+        <wsc:body>#[payload]</wsc:body>
+        <wsc:headers>
+          #[
+          %dw 2.0
+          output application/xml
+          ns con http://service.soap.clients.namespace/
+          ---
+          "headers": {
+              con#user: "admin",
+              con#pass: "textpassword"
+          }]
+        </wsc:header>
+    </wsc:message>
+</wsc:consume>
+----
+
+==== Attachments Parameter
+
+The `attachments` parameter enables you to bind attachments to the SOAP message. To create attachments for transport over SOAP, declare a DataWeave script in which each entry represents an attachment, and the entry value provides the content of the attachment.
+
+The following XML example shows a DataWeave expression inside the `attachments` parameter that declares a new attachment called `clientsJson`. The attachment value content is stored in the `jsonFile` variable. This variable could be set from a `file:read` operation:
+
+[source,xml,linenums]
+----
+<wsc:consume config-ref="config" operation="addClients">
+    <wsc:message>
+        <wsc:body>#[payload]</wsc:body>
+        <wsc:attachments>
+          #[{ clientsJson: vars.jsonFile } ]
+        </wsc:attachments>
+    </wsc:message>
+</wsc:consume>
+----
+
+=== Output
+
+The output of the Consume operation represents an incoming SOAP message that contains the same elements that the Message parameter has, and you can access each part of it.
+
+The following XML example stores:
+
+* The content of the body in a `soap.body` variable
+* The content of the header called `auth` in a `soap.auth.header` variable
+* The content of an attachment called `json` in a `soap.attachment.json` variable
+
+[source,xml,linenums]
+----
+<flow name="output">
+  <wsc:consume config-ref="config" operation="addClients">
+      <wsc:message>
+          <wsc:body>#[payload]</wsc:body>
+      </wsc:message>
+  </wsc:consume>
+  <set-variable name="soap.body" value="#[payload.body]">
+  <set-variable name="soap.auth.header" value="#[payload.headers.auth]">
+  <set-variable name="soap.attachment.json" value="#[payload.attachments.json]">
+</flow>
+----
+
+=== Attributes
+
+When consuming a Web service operation, you might be interested not only in the response content but also in the metadata of the underlying transport used to dispatch the messages. For example, when you use
+HTTP, attributes carry HTTP headers bounded to the HTTP request (`content-length`, `status`, and so on).
+
+The Web Service Consumer Connector uses Mule Message Attributes to access this information.
+
+
+== Next Step
+
+After you configure a global element and connection information, configure the other fields for the connector. See xref:web-service-consumer-config-topics.adoc[Additional Configuration Information] for more configuration steps.
```

Comment:
```suggestion
After you configure a global element and provide connection information, configure the other fields for the connector. See Additional Configuration Information for more configuration steps.
```
...although I would like to see the template change the wording of this subtopic title going forward....

Reply:
You can move the link to See Also.